test/e2e/util.go

```diff
@@ -30,6 +32,7 @@ type providerTestHelper interface {
 	platform() string
 	makeCredentialsSecret(namespace string) *corev1.Secret
 	buildExternalDNS(name, zoneID, zoneDomain string, credsSecret *corev1.Secret) operatorv1alpha1.ExternalDNS
+	buildRouteExternalDNS(name, routeName, zoneID, zoneDomain string) operatorv1alpha1.ExternalDNS
```

Comment:
```suggestion
	buildOpenShiftExternalDNS(name, routeName, zoneID, zoneDomain string) operatorv1alpha1.ExternalDNS
```
As it's not only about the route source.